### ISSUE TITLE
Frameless minimum sized window is not restored to previous size after maximize and unmaximize

### ISSUE THREAD

**Issue by christian-judt** [electron/electron#22440]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 7.1.2 - 9.0.0-beta.3 (and presumably later versions until this bug is fixed)
* **Operating System:**
  * Windows 10 pro (1909)
* **Last Known Working Electron version:**
  * 7.1.1

### Preconditions

* Window is frameless
* Window has minimum size
* Window size is minimum size

### Expected behavior

* Window is maximized
* Window is unmaximized
* Window has the same size as before the maximization

### Actual Behavior
* Window is maximized
* Window is unmaximized
* Window is slightly higher and wider than before the maximization

### To Reproduce

```
git clone -b frameless-minsize-maximize-unmaximize-bug https://github.com/christian-judt/electron-quick-start.git
cd electron-quick-start
npm install
npm start
```

### Additional Information

* This seems to be a reappearing problem. (for the last occurence see #13533)
* A similar problem with "minimize-restore" instead of "maximize-unmaximize" has appeared in Electron Version 7.1.10. (see #22393)

### Other maybe related issues

* [Windows] Frameless window size increase on a maximize, unmaximize sequence #2498
* Windows: Restore (after minimize) to wrong window size #7951
* maximize, minimize, restore incorrect size #15702
* Electron 6 - restore after minimize has wrong window size #19816

**Comment by aleksey-hoffman** [comment]
I reproduced it  in version `8.0.3`. I see the same behavior:

- Resized window to the defined minimum (500x300)
- Maximize, un-maximize
- The window size changes to 515x315

```JS
mainWindow = new BrowserWindow({
    title: "Title",
    width: 1280,
    height: 720,
    minWidth: 500,
    minHeight: 300,
    frame: false,
    transparent: false,
    resizable: true
})
```

**Comment by PsykickSam** [comment]
I made a solution for window resizing, I don't know the problem but It may cause for minWidth and minHeight so I remove the minWidth and minHeight from BrowserWindow and add an resize event, 

mainWindow.on("resize", () => {
    const w = mainWindow.getSize()[0];
    const h = mainWindow.getSize()[1];

    if (w < 800 || h < 600) {
      mainWindow.setSize(800, 600);
    } 
  });

It worked for me.

**Comment by Pettor** [comment]
I have similar issue when using `frame: false` and `resizable: false` on Windows. Whenever the window is hidden (minimized or closed to taskbar) and I open again the size is incorrect (a pixels wider and higher).

**Comment by christian-judt** [comment]
I noticed the structurally related Issue #22393 has been closed in #25014. Ist this change also affecting this Issue?
